Encoder/JWTEncoderInterface.php

```diff
@@ -14,21 +14,23 @@
 {
     /**
      * @param array $data
+     * @param array $header
      *
      * @return string the encoded token string
      *
      * @throws JWTEncodeFailureException If an error occurred while trying to create
      *                                   the token (invalid crypto key, invalid payload...)
      */
-    public function encode(array $data);
+    public function encode(array $data, array $header = []);
 
     /**
      * @param string $token
+     * @param array  $header
      *
      * @return array
      *
      * @throws JWTDecodeFailureException If an error occurred while trying to load the token
      *                                   (invalid signature, invalid crypto key, expired token...)
      */
-    public function decode($token);
+    public function decode($token, array &$header = []);
```

Comment:
The idea is to find a way to return the header after the decoding process, not to modify it.
But I have no specific use case and I will remove it.